e2e/tests/ui-driven/src/create-flow.spec.ts

```diff
@@ -177,6 +179,54 @@ test.describe("Flow creation, publish and preview", () => {
     await expect(previewLink).toBeVisible();
   });
 
+  test("Can add an external portal", async ({
+    browser,
+  }: {
+    browser: Browser;
+  }) => {
+    const page = await createAuthenticatedSession({
+      browser,
+      userId: context.user!.id!,
+    });
+
+    await page.goto(`/${context.team.slug}`);
+
+    const editor = new PlaywrightEditor(page);
+
+    page.on("dialog", (dialog) =>
+      dialog.accept(externalPortalServiceProps.name),
+    );
+    await editor.addNewService();
+
+    // update context to allow new flow to be torn down
+    context.flows?.push({ ...externalPortalServiceProps });
+
+    await createQuestionWithOptions(
+      page,
+      editor.firstNode,
+      externalPortalFlowData.title,
+      externalPortalFlowData.answers,
+    );
+    await expect(editor.nodeList).toContainText([
+      externalPortalFlowData.title,
+      externalPortalFlowData.answers[0],
+      externalPortalFlowData.answers[1],
+    ]);
+
+    await publishService(page);
```

Comment:
question: why do we publish the service twice in this test? Can we not just create all the components and then publish/turn online at the end? I guess this would remove the need to call `navigateToService` as well.

Reply:
Not sure if we're talking about the same thing 😅 I think it's fair enough having a separate portal test like this. I was referring to the fact that in this test ('can create external portal') you have:
```js
createQuestionWithOptions()

publishService() // <--- this is the publish I have issue with! 
// Why do we publish after creating the question as well as at the end (after creating the portal)?

turnServiceOnline() // I think this should be moved to the bottom with the publish

navigateToService()

createExternalPortal()

publishService() // I think this should be the only publish
```

Happy to chat if you prefer 😁

Reply:
Ahh because we need to publish the external portal service, then add that external portal to the main service and publish that as a change.

1. Create **External Portal service**
2. Publish **External Portal service**
3. Navigate to **Main service**
4. Add External Portal component to **Main service**
5. Publish changes to **Main service**

Otherwise it won't pick up the external portal when we answer the questions in the published link?

Reply:
Ah ok that makes sense! Maybe just a comment next to each publish to say which service is being published then :)

Reply:
Yeah that's sick!